src/components/molecules/VenueChat/VenueChat.tsx

```diff
@@ -16,15 +9,17 @@ import { VENUE_CHAT_AGE_DAYS } from "settings";
 import { getDaysAgoInSeconds, roundToNearestHour } from "utils/time";
 import { currentVenueSelectorData, venueChatsSelector } from "utils/selectors";
 
+import { useDispatch } from "hooks/useDispatch";
 import useRoles from "hooks/useRoles";
 import { useUser } from "hooks/useUser";
 import { useSelector } from "hooks/useSelector";
 import { useVenueId } from "hooks/useVenueId";
 
-import { ChatContext, chatSort } from "components/context/ChatContext";
+import { chatSort } from "utils/chat";
```

Comment:
This is interesting. Not sure I agree about the suggestion, we should hop on zoom and talk through it.

`chatSort` is not grammatically intended to say that it sorts the chat (that is able to be inferred from context). It's also a sorting predicate, so `Sort` should not be used as a verb, because it doesn't sort - it only describes how to sort. But there's still a `Sort` in there so what's it for? The `Sort` in `chatSort` serves to disambiguate it from other sorting predicates. Including `Sort` in `chatSort` tells the reader: "this is the chat sort predicate, you can expect there to be others elsewhere in the code".

Something called `sortChat` would need to be more self-contained to earn that title. It's the difference between:

```
export const chatSort: (a: BaseChatMessage, b: BaseChatMessage) => number = (
  a: BaseChatMessage,
  b: BaseChatMessage
) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf());
```

And:

```
const sortChat(chat: ???): ??? =>
 chat.sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf()));
```

The latter performs the action, whereas the former is only the predicate.

There's another problem with `sortChat` which is that we're calling these `chats` in the code, so the pluralization doesn't work.

My leaning would be to keep `chatSort` but happy to have a chat and see what you think. And thanks for stimulating the discussion!

Reply:
Valid points. Agreed